### PR TITLE
Return NotFound on KEB instance updates when instance is missing

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**

Changes proposed in this pull request:

- KEB returns 5xx error on instance update method when instance is not found in db. This proposes a simple change to return 404 instead.

**Related issue(s)**

https://github.com/orgs/kyma-project/projects/10#card-60954090

### CODE REVIEW

File: components/kyma-environment-broker/internal/broker/instance_update.go [hunk 1]
```diff
@@ -45,7 +48,10 @@ func (b *UpdateEndpoint) Update(ctx context.Context, instanceID string, details
 	logger.Infof("Update asyncAllowed: %v", asyncAllowed)
 
 	instance, err := b.instanceStorage.GetByID(instanceID)
-	if err != nil {
+	if err != nil && dberr.IsNotFound(err) {
+		logger.Errorf("unable to get instance: %s", err.Error())
+		return domain.UpdateServiceSpec{}, apiresponses.NewFailureResponse(err, http.StatusNotFound, fmt.Sprintf("could not execute update for instanceID %s", instanceID))
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/broker/instance_update.go:53]: I tried to take an inspiration from:
https://github.com/kyma-project/control-plane/blob/634f84ee59828dc72c6a4aae6c3632ab11464e9b/components/kyma-environment-broker/internal/broker/instance_deprovision.go#L56

but I think the if-else block reads a bit easier than the switch case.

**Review by wozniakjan — COMMENTED** [review]
wondering if we should potentially update following as well:
https://github.com/kyma-project/control-plane/blob/634f84ee59828dc72c6a4aae6c3632ab11464e9b/components/kyma-environment-broker/internal/broker/instance_get.go#L31-L34
https://github.com/kyma-project/control-plane/blob/634f84ee59828dc72c6a4aae6c3632ab11464e9b/components/kyma-environment-broker/internal/broker/instance_create.go#L93-L95
https://github.com/kyma-project/control-plane/blob/634f84ee59828dc72c6a4aae6c3632ab11464e9b/components/kyma-environment-broker/internal/broker/instance_create.go#L98-L100

**Review by piotrmiskiewicz — DISMISSED** [review]
lgtm

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**